marconi-tutorial/src/Marconi/Tutorial/HttpServer.hs

```diff
@@ -35,6 +40,42 @@ type RpcMethod = RpcAddressCountMethod
 type RpcAddressCountMethod =
   JsonRpc
     "getAddressCount"
+    AddressCountQuery
     String
-    String
-    Int
+    (Core.Result AddressCountQuery)
+
+{----------------------
+   API implementation
+-----------------------}
+
+runHttpServer :: ReaderT Env IO ()
+runHttpServer = do
+  env <- ask
+  let cliOptions = view envCliArgs env
+  let httpSettings = maybe defaultSettings (flip setPort defaultSettings) (optionsHttpPort cliOptions)
+  liftIO $ runSettings httpSettings (httpApp env)
+
+httpApp
+  :: Env
+  -> Application
+httpApp env = serve (Proxy @API) (httpServer env)
+
+httpServer
+  :: Env
+  -> Server API
+httpServer = getAddressCountHttpHandler
+
+getAddressCountHttpHandler
+  :: Env
+  -> AddressCountQuery
+  -> Handler (Either (JsonRpcErr String) (Core.Result AddressCountQuery))
+getAddressCountHttpHandler env q = do
+  let indexerVar =
+        env
+          ^. envIndexers
+            . addressCountIndexerEnv
+            . addressCountIndexerVar
+  liftIO $
+    readIORef indexerVar >>= readMVar >>= \indexer -> do
+      r <- Core.query Core.genesis q indexer
```

Comment:
Promote the use of `queryLatest` in the tutorial maybe?

Reply:
True, but the user experience is better with `query` right now. With `queryLatest`, I get a `MonadError` that I need to handle.

Reply:
Indeed, but you don't have this error in the general case at the cost of accepting the risk of asynchronous errors sent by sqlite in the `Queryable` instance.
I know it adds complexity but I think we should promote explicit errors here.

Reply:
True. Also, I think I misinterpreted `query`. By using `query genesis`, it should basically return nothing, right? So, using `query` with `getLastSyncPoint` should indeed yield a `MonadError`.

Reply:
Yes, it should be this way, although, we don't have any mean to enforce it: an indexer implementation could totally ignore the parameter. If you want to yield the errors, there is some helper in the `SQLiteIndexer` package. Promoting them is probably a good idea.